packages/astro/src/integrations/features-validation.ts

```diff
@@ -90,69 +85,91 @@ export function validateSupportedFeatures(
 		() => Object.keys(config?.env?.schema ?? {}).length !== 0,
 	);
 
+	validationResult.sharpImageService = validateSupportKind(
+		sharpImageService,
+		adapterName,
+		logger,
+		'sharp',
+		() => config?.image.service.entrypoint === 'astro/assets/services/sharp',
+	);
+
 	return validationResult;
 }
 
+export function unwrapSupportKind(supportKind?: AdapterSupport): AdapterSupportsKind | undefined {
+	if (!supportKind) {
+		return undefined;
+	}
+
+	return typeof supportKind === 'object' ? supportKind.support : supportKind;
+}
+
+export function getSupportMessage(supportKind: AdapterSupport): string | undefined {
+	return typeof supportKind === 'object' ? supportKind.message : undefined;
+}
+
 function validateSupportKind(
-	supportKind: AdapterSupportsKind,
+	supportKind: AdapterSupport,
 	adapterName: string,
 	logger: Logger,
 	featureName: string,
 	hasCorrectConfig: () => boolean,
 ): boolean {
-	if (supportKind === STABLE) {
-		return true;
-	} else if (supportKind === DEPRECATED) {
-		featureIsDeprecated(adapterName, logger, featureName);
-	} else if (supportKind === EXPERIMENTAL) {
-		featureIsExperimental(adapterName, logger, featureName);
-	}
+	const supportValue = unwrapSupportKind(supportKind);
+	const message = getSupportMessage(supportKind);
 
-	if (hasCorrectConfig() && supportKind === UNSUPPORTED) {
-		featureIsUnsupported(adapterName, logger, featureName);
+	if (!supportValue) {
 		return false;
-	} else {
-		return true;
 	}
-}
 
-function featureIsUnsupported(adapterName: string, logger: Logger, featureName: string) {
-	logger.error(
-		'config',
-		`The adapter ${adapterName} doesn't currently support the feature "${featureName}".`,
-	);
-}
-
-function featureIsExperimental(adapterName: string, logger: Logger, featureName: string) {
-	logger.warn(
-		'config',
-		`The adapter ${adapterName} provides experimental support for "${featureName}". You may experience issues or breaking changes until this feature is fully supported by the adapter.`,
-	);
-}
+	if (supportValue === AdapterFeatureStability.STABLE) {
+		return true;
+	} else if (hasCorrectConfig()) {
```

Comment:
The previous check here was weird, it only checked if the config was correct for unsupported values, but you need to always check it, otherwise you get warning for every experimental feature, even the ones you don't use?